### PR TITLE
JBTM-3510 Replace new Integer(int) with valueOf(int) and also for oth…

### PR DESCRIPTION
**Pull request by RanabirChakraborty**
Issue: https://issues.redhat.com/browse/JBTM-3510

### CODE REVIEW

**Comment by RanabirChakraborty** [comment]
TOMCAT, JACOCO, LRA config got failed for jdk8, tho' I'm not sure but looking at the errors I don't think it was affected by the PR changes.

File: ArjunaJTS/orbportability/classes/com/arjuna/orbportability/common/ant/IDLCompiler.java [hunk 1]
```diff
@@ -84,12 +84,12 @@ public IDLCompiler()
 
     public final void setDebug(String debug)
     {
-        _debugOn = new Boolean(debug).booleanValue();
+        _debugOn = Boolean.valueOf(debug).booleanValue();
```
Comment (mmusgrov) [inline ArjunaJTS/orbportability/classes/com/arjuna/orbportability/common/ant/IDLCompiler.java:87]: You don't need to unbox the boolean value, ie `_debugOn = Boolean.valueOf(debug)` is sufficient.
Similarly elsewhere in this PR.

Also `Boolean.valueOf(debug)` assigns a boxed value to the primitive `_debugOn` so incurs unboxing.

So to summarise the best approach is to just do `_debugOn = Boolean.parseBoolean(debug);` 

Similarly elsewhere in this PR (ie I won't repeat the same comment).

File: STM/src/main/java/org/jboss/stm/internal/proxy/LockManagerProxy.java [hunk 2]
```diff
@@ -1035,21 +1035,21 @@ private boolean unpackObjectType (final Field afield, InputObjectState os)
             // TODO arrays
             
             if (afield.getType().equals(Boolean.class))
-                afield.set(_theObject, new Boolean(os.unpackBoolean()));
+                afield.set(_theObject, Boolean.valueOf(os.unpackBoolean()));
```
Comment (mmusgrov) [inline STM/src/main/java/org/jboss/stm/internal/proxy/LockManagerProxy.java:1038]: You don't need to box the value before passing it to `afield.set`. So `afield.set(_theObject, os.unpackBoolean());` is sufficient. Similarly for the other changes in this routine and in other places where the same unnecessary boxing has been done in this PR.

File: STM/src/test/java/org/jboss/stm/internal/proxy/LockManagerProxyUnitTest.java [hunk 3]
```diff
@@ -67,8 +67,8 @@ public String toString ()
     {
         public ExtendedLockable ()
         {
-            _isState = new Double(1.234);
-            _isNotState = new Integer(5678);
+            _isState = Double.valueOf(1.234);
```
Comment (mmusgrov) [inline STM/src/test/java/org/jboss/stm/internal/proxy/LockManagerProxyUnitTest.java:70]: You don't need to box the primitive and then assign it to a primitive, ie `_isState = 1.234;` is sufficient. Similarly where the same has been done elsewhere in this PR.

File: XTS/WS-C/dev/src/com/arjuna/wsc11/messaging/ActivationCoordinatorProcessorImpl.java [hunk 4]
```diff
@@ -65,7 +65,7 @@
                 try
                 {
                     final Expires expiresElement = createCoordinationContext.getExpires() ;
-                    final Long expires = (expiresElement == null ? null : new Long(expiresElement.getValue())) ;
+                    final Long expires = (expiresElement == null ? null : Long.valueOf(expiresElement.getValue())) ;
```
Comment (mmusgrov) [inline XTS/WS-C/dev/src/com/arjuna/wsc11/messaging/ActivationCoordinatorProcessorImpl.java:68]: Unnecessary boxing, `expiresElement.getValue()` is sufficient. Similarly elsewhere in this PR.

File: XTS/WSTX/classes/com/arjuna/mwlabs/wst11/at/remote/UserTransactionImple.java [hunk 5]
```diff
@@ -288,7 +288,7 @@ protected final Context startTransaction(int timeout, TxContextImple current)
             //InputStream inputStream = Thread.currentThread().getContextClassLoader().getResourceAsStream("/foo.properties");
 
 
-            final Long expires = (timeout > 0 ? new Long(timeout) : null) ;
+            final Long expires = (timeout > 0 ? Long.valueOf(timeout) : null) ;
```
Comment (mmusgrov) [inline XTS/WSTX/classes/com/arjuna/mwlabs/wst11/at/remote/UserTransactionImple.java:291]: Unnecessary boxing, use `(long) timeout` instead. Similarly elsewhere in this PR.

File: XTS/localjunit/WSTFSC07-interop/src/main/java/com/jboss/transaction/wstf/test/TestServlet.java [hunk 6]
```diff
@@ -200,7 +200,7 @@ private int getLogCount(final HttpSession session)
     {
         final Object logCountObject = session.getAttribute(TestConstants.ATTRIBUTE_LOG_COUNT) ;
         final int logCount = (logCountObject == null ? 1 : ((Integer)logCountObject).intValue() + 1) ;
-        session.setAttribute(TestConstants.ATTRIBUTE_LOG_COUNT, new Integer(logCount)) ;
+        session.setAttribute(TestConstants.ATTRIBUTE_LOG_COUNT, Integer.valueOf(logCount)) ;
```
Comment (mmusgrov) [inline XTS/localjunit/WSTFSC07-interop/src/main/java/com/jboss/transaction/wstf/test/TestServlet.java:203]: Unnecessary boxing (similarly elsewhere in this PR).

File: XTS/localjunit/unit/src/test/java/com/arjuna/wsc/tests/arq/ActivationTest.java [hunk 7]
```diff
@@ -92,7 +92,7 @@ public void testRequestWithExpiresWithoutCurrentContext()
             {
         final String messageId = "testRequestWithExpiresWithoutCurrentContext" ;
         final String coordinationType = TestUtil.COORDINATION_TYPE ;
-        final Long expires = new Long(123456L) ;
+        final Long expires = Long.valueOf(123456L) ;
```
Comment (mmusgrov) [inline XTS/localjunit/unit/src/test/java/com/arjuna/wsc/tests/arq/ActivationTest.java:95]: Unecessary boxing

**Review by mmusgrov — CHANGES_REQUESTED** [review]
There are quite a few examples where the boxing and unboxing are unnecessary.

**Comment by RanabirChakraborty** [comment]
@mmusgrov did the changes accordingly. Can you please review this once :)

**Review by mmusgrov — APPROVED** [review]
@RanabirChakraborty thanks for the fixes. The PR looks good and I will merge it.

**Comment by RanabirChakraborty** [comment]
Thanks! :)

**Pull request merged**